tools/SetupFlow/DevHome.SetupFlow/Models/CloneRepoTask.cs

```diff
@@ -271,6 +280,9 @@ IAsyncOperation<TaskFinishedState> ISetupTask.Execute()
                 }
             }
 
+            _host.GetService<RepositoryManagementDataAccessService>()

```

Comment:
There is no try-catch at this point in the code, if the DB has a problem, it's going to throw a likely unhandled exception (I didn't look further up the chain).

Also, assuming the above issue is resolved, what happens with the torn state when the clone is successful but there's a database writing failure? We will have partial success here - it will be successfully cloned, but we may not know about it in the db.

Reply:
> There is no try-catch at this point in the code, 

Correct.  This PR will not have a try/catch.  It will  in the future though.  By future, I mean the next PR, or the one after that.  I will not release this product without exception handling in place.

> hat happens with the torn state when the clone is successful but there's a database writing failure

That is a super interesting question and not one I have thought of.  We could attempt the save again.  Maybe log the error and tell the user "Something happened.  Please click 'Add Existing repository' for repository X" maybe something else.

In all cases, your question still stands.  What happens when the DevHome action is successful, but the database action is not.  I will think on this.